api/src/main/java/net/md_5/bungee/api/connection/ProxiedPlayer.java

```diff
@@ -339,4 +340,36 @@ public enum MainHand
      */
     @Deprecated
     Scoreboard getScoreboard();
+
+    /**
+     * Retrieves a cookie from this player.
+     *
+     * @param cookie the resource location of the cookie, for example "bungeecord:my_cookie"
+     *
+     * @return a {@link CompletableFuture} that will be completed when the Cookie response is received
+     * Only useable for 1.20.5 clients or newer,
+     * if the cookie is not set in the client, the {@link CompletableFuture} will complete with a null value
+     */
+    CompletableFuture<byte[]> retrieveCookie(String cookie);
+
+
+    /**
+     * Stores a cookie in this player's client.
+     *
+     * @param cookie the resource location of the cookie, for example "bungeecord:my_cookie"
+     * @param data the data to store in the cookie
+     *
+     * Only useable for 1.20.5 clients or newer
+     */
+    void storeCookie(String cookie, byte[] data);
+
+    /**
+     * Transfers this player to a different server specified by host and port.
```

Comment:
Please be more specific that it asks the client to transfer, it's not a bungee transfer